Mage.Sets/src/mage/cards/c/CuratorsWard.java

```diff
@@ -0,0 +1,120 @@
+
+import java.util.UUID;
+import mage.constants.SubType;
+import mage.abilities.Ability;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.common.AttachEffect;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.abilities.effects.common.continuous.GainAbilityAttachedEffect;
+import mage.constants.Outcome;
+import mage.target.TargetPermanent;
+import mage.abilities.keyword.EnchantAbility;
+import mage.abilities.keyword.HexproofAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.AttachmentType;
+import mage.constants.CardType;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.events.ZoneChangeEvent;
+import mage.game.permanent.Permanent;
+
+/**
+ *
+ * @author keelahnkhan
+ */
+public class CuratorsWard extends CardImpl {
+
+    public CuratorsWard(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{2}{U}");
+        
+        this.subtype.add(SubType.AURA);
+
+        // Enchant permanent
+        TargetPermanent auraTarget = new TargetPermanent();
+        this.getSpellAbility().addTarget(auraTarget);
+        this.getSpellAbility().addEffect(new AttachEffect(Outcome.AddAbility));
+        Ability ability = new EnchantAbility(auraTarget.getTargetName());
+        this.addAbility(ability);
+
+        // Enchanted permanent has hexproof.
+        ability.addEffect(new GainAbilityAttachedEffect(HexproofAbility.getInstance(), AttachmentType.AURA).setText("Enchanted permanent has hexproof"));
+        
+        // When enchanted permanent leaves the battlefield, if it was historic, draw two cards.
+        this.addAbility(new CuratorsWardTriggeredAbility());
+    }
+
+    public CuratorsWard(final CuratorsWard card) {
+        super(card);
+    }
+
+    @Override
+    public CuratorsWard copy() {
+        return new CuratorsWard(this);
+    }
+}
+
+class CuratorsWardTriggeredAbility extends TriggeredAbilityImpl {
+
+    public CuratorsWardTriggeredAbility() {
+        super(Zone.BATTLEFIELD, new DrawCardSourceControllerEffect(2).setText("When enchanted permanent leaves the battlefield, if it was historic, draw two cards."), false);
+    }
+
+    public CuratorsWardTriggeredAbility(CuratorsWardTriggeredAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.ZONE_CHANGE;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        ZoneChangeEvent zEvent = (ZoneChangeEvent) event;
+        if (zEvent.getFromZone() == Zone.BATTLEFIELD) {
+            Permanent enchanted = game.getPermanentOrLKIBattlefield(event.getTargetId());
+            if (enchanted != null && enchanted.getAttachments().contains(sourceId) && enchanted.isHistoric()) {
```

Comment:
Did you test whether this works when both the aura and the enchanted permanent leave at the same time?

Reply:
I believe so and tested again after you pointed this out to be sure (Akroma's Vengeance was what I tried), the ability still triggers as long as the enchanted permanent is historic. This is how it should work right?

Reply:
Yeah, I just wanted to confirm that based on other similar cards.